### PR TITLE
[CI] Re-enable API and Generator diff. 

### PR DESCRIPTION
**Pull request by mandel-macaque**
Use the jenkins script as a base to get the PAI & generator from stable
diff back to the CI. Comment is not perfect, but does provide the
required information. The comment can be improve in a later commit since
this is getting too large.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Sample commit from the tests: https://github.com/xamarin/xamarin-macios/commit/c8460333ead81d195d60adc7b16733f81c42f032#commitcomment-49333692

We should get a similar one here nevertheless.

**Review by spouliot — COMMENTED** [review]
✅ API Diff (from PR only) (no change) ✅ Generator Diff (no change)

The output^ would be nicer on separate lines

✅ API Diff (from PR only) (no change) 
✅ Generator Diff (no change)

**Comment by mandel-macaque** [comment]
@spouliot I know: "The comment can be improve in a later commit since
this is getting too large." ;)

**Review by spouliot — COMMENTED** [review]
@mandel-macaque do a small binding change in a commit

that will make API diff and generator diff generate something (non empty)

and then revert it - that will make the reports empty again

and proving the PR works as intended :)

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/templates/build/upload-vsdrops.yml [hunk 1]
```diff
@@ -41,3 +41,13 @@ steps:
     sourcePath: $(STABLE_APIDDIFF_PATH)
     detailedLog: true
     usePat: true
+
+- task: ms-vscs-artifact.build-tasks.artifactDropTask-1.artifactDropTask@0
+  displayName: 'Publish API & Generator comparisonn to Artifact Services Drop'
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/build/upload-vsdrops.yml:46]:
```suggestion
  displayName: 'Publish API & Generator comparison to Artifact Services Drop'
```

File: tools/devops/automation/scripts/bash/compare.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+BUILD_URL="$SYSTEM_TEAMFOUNDATIONCOLLECTIONURI$SYSTEM_TEAMPROJECT/_build/index?buildId=$BUILD_BUILDID&view=ms.vss-test-web.test-result-details"
+URL_PREFIX="$VSDROPSPREFIX/$BUILD_BUILDNUMBER/$BUILD_BUILDID/apigeneratordiff/;/"
+API_URL="$URL_PREFIX/api-diff.html"
+GENERATOR_URL="$URL_PREFIX/generator-diff/index.html"
+
+# env var should have been defined by the CI
+if test -z "$XAM_TOP"; then
+    echo "Variable XAM_TOP is missing."
+    exit 1
+fi
```
Comment (rolfbjarne) [inline tools/devops/automation/scripts/bash/compare.sh:33]: This contradicts the documentation at the top: `To compare a specific commit to the previous one, just run this script (./compare.sh).`

In fact, the changes in this script looks like it won't be easy to run locally. Did you test running it locally to just compare with the previous commit?

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tools/devops/automation/scripts/bash/compare.sh (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/automation/scripts/bash/compare.sh:33]: Need to fix that

File: tools/devops/automation/scripts/bash/vsts-compare.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# set the env vars that are required for the compare to work. XAM_TOP is not set because
+# the yaml takes care of that.
+export WORKSPACE="$BUILD_ARTIFACTSTAGINGDIRECTORY"
+echo "GH PR: $ghprbPullId"
```
Comment (rolfbjarne) [inline tools/devops/automation/scripts/bash/vsts-compare.sh:4]: You're not calling compare.sh here.

File: tools/devops/automation/scripts/bash/vsts-compare.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# set the env vars that are required for the compare to work. XAM_TOP is not set because
```
Comment (rolfbjarne) [inline tools/devops/automation/scripts/bash/vsts-compare.sh:1]: Maybe start with `#!/bin/bash -e`?

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by dalexsoto** [comment]
Do not merge label per Manuel's request

**Comment by mandel-macaque** [comment]
@spouliot success: https://github.com/xamarin/xamarin-macios/pull/11244#issuecomment-828934916 there are two improvements I want to do, but we are good to land AFAIK

**Pull request merged**